### PR TITLE
Sameera/schooldetailview

### PR DESCRIPTION
**Pull request by sameerav**

### CODE REVIEW

**Comment by amillman** [comment]
can you zoom in the mapview a little bit more? like how it is in the checkin view. can you also make the view not interactable (so no dragging)

File: sage-ios/SAGE/SAGE/AdminOperations.swift [hunk 1]
```diff
@@ -99,6 +99,38 @@ class AdminOperations {
         
     }
     
+    static func loadSchool(id: Int, completion: ((School) -> Void), failure: (String) -> Void) {
+        let requestURL = StringConstants.kEndpointBaseURL + "/schools/" + String(id)
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AdminOperations.swift]: you can make a method in StringConstants where you pass in the id and itll return the correct url. so its like StringConstants.kDetailSchoolURL(id) or osmething like that

File: sage-ios/SAGE/SAGE/AdminOperations.swift [hunk 2]
```diff
@@ -99,6 +99,38 @@ class AdminOperations {
...
+            var schoolSwiftDict = [String: AnyObject]()
+            for key in schoolDict.allKeys {
+                schoolSwiftDict[key as! String] = schoolDict[key as! String]
+            }
+            let school = School(propertyDictionary: schoolSwiftDict)
+            var students = [User]()
+            for user in userDict {
+                let userDict = user as! NSMutableDictionary
+                var userSwiftDict = [String: AnyObject]()
+                for key in userDict.allKeys {
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AdminOperations.swift]: ? ?? ? what dis

File: sage-ios/SAGE/SAGE/BrowseMentorsTableViewCell.swift [hunk 3]
```diff
@@ -77,6 +77,10 @@ class BrowseMentorsTableViewCell: UITableViewCell {
         self.mentorName.setWidth(CGRectGetMinX(self.totalHours.frame) - mentorSchoolX - UIConstants.textMargin)
         self.schoolName.font = UIFont.getDefaultFont(14)
         self.schoolName.textColor = UIColor.secondaryTextColor
+        
+        if self.schoolName.text == nil {
+            self.mentorName.centerVertically()
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/BrowseMentorsTableViewCell.swift:89]: yee yee

File: sage-ios/SAGE/SAGE/BrowseSchoolsDetailViewController.swift [hunk 4]
```diff
@@ -8,22 +8,89 @@
 
 import UIKit
 
-class BrowseSchoolsDetailViewController: UIViewController {
+class BrowseSchoolsDetailViewController: UITableViewController {
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/BrowseSchoolsDetailViewController.swift]: rename to SchoolDetailViewController

File: sage-ios/SAGE/SAGE/BrowseSchoolsTableViewCell.swift [hunk 5]
```diff
@@ -10,15 +10,59 @@ import UIKit
 
 class BrowseSchoolsTableViewCell: UITableViewCell {
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/BrowseSchoolsTableViewCell.swift]: rename to SchoolTableViewCell

File: sage-ios/SAGE/SAGE/BrowseSchoolsTableViewCell.swift [hunk 6]
```diff
@@ -10,15 +10,59 @@ import UIKit
...
+    
+    override func layoutSubviews() {
+        
+        super.layoutSubviews()
+        
+        self.contentView.addSubview(self.numStudents)
+        self.numStudents.sizeToFit()
+        self.numStudents.centerVertically()
+        self.numStudents.setX(self.frame.width - self.numStudents.frame.width - UIConstants.sideMargin)
+        self.numStudents.textColor = UIColor.secondaryTextColor
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/BrowseSchoolsTableViewCell.swift]: put non-layout stuff in a setup method that only gets called once

File: sage-ios/SAGE/SAGE/BrowseSchoolsViewController.swift [hunk 7]
```diff
@@ -86,12 +86,20 @@ class BrowseSchoolsViewController: UITableViewController {
...
+    
     override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
         let school = self.schools![indexPath.row]
-        let vc = BrowseSchoolsDetailViewController(school: school)
+        let vc = BrowseSchoolsDetailViewController()
         if let topItem = self.navigationController!.navigationBar.topItem {
             topItem.backBarButtonItem = UIBarButtonItem(title: "", style: .Plain, target: nil, action: nil)
         }
+        vc.configureWithSchool(school)
+        AdminOperations.loadSchool(school.id, completion: { (school) -> Void in
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/BrowseSchoolsViewController.swift]: this should be called in the detail view controller itself

File: sage-ios/SAGE/SAGE/AdminOperations.swift (hunk 2 above)
Comment (sameerav) [inline sage-ios/SAGE/SAGE/AdminOperations.swift]: dis to convert from nsmutabledictionary to swift dictionary so we can pass into init(propertyDictionary:) methods

Comment (amillman) [inline sage-ios/SAGE/SAGE/AdminOperations.swift]: u sure thats needed? we never needed that before

Comment (sameerav) [inline sage-ios/SAGE/SAGE/AdminOperations.swift]: nah we always did that because property dictionary is supposed to be a swift dictionary

Comment (sameerav) [inline sage-ios/SAGE/SAGE/AdminOperations.swift]: oh yus you are right refactored all of the things in admin operations

**Pull request merged**